firebase.agent.nut

```diff
@@ -383,8 +383,42 @@ class Firebase {
         }.bindenv(this);
     }
 
+    function _isValidJson(text) {
+        try {
+            http.jsondecode(text);
+        } catch (err) { 
+            return false; 
+        }
+        return true; 
+    }
+
     // parses event messages
-    function _parseEventMessage(text) {
+    function _parseEventMessage(input) {
+        local text = _bufferedInput + input;
+        local dataIndex = text.find("data");
+        local bracketIndex = text.find("{");
+
+        if (dataIndex && dataIndex < bracketIndex) {
```

Comment:
What does `"data" word before Json` mean? It's probably easier to discuss it offline.